shpopen.c

```diff
@@ -100,65 +72,65 @@ void SHPAPI_CALL SHPWriteHeader(SHPHandle psSHP)
     uint32_t i32 = psSHP->nFileSize / 2; /* file size */
     ByteCopy(&i32, abyHeader + 24, 4);
 #if !defined(SHP_BIG_ENDIAN)
-    SwapWord(4, abyHeader + 24);
+    SHP_SWAP32(STATIC_CAST(uint32_t *, STATIC_CAST(void *, abyHeader + 24)));
```

Comment:
Sorry but all those casts look super ugly and make the code hardly readable

Reply:
I know. First I need the GDAL test uiste result, then I will update,

Reply:
> I know. First I need the GDAL test suite result, then I will update,

well, you can easily test that yourself. clone OSGeo/GDAL, replace the shapefil.h, shpopen.c, dbfopen.c, shptree.c and sbnsearch.c files in ogr/ogrsf_frmts/shape and look at the CI results